tf-modules/cumulus-message-adapter/outputs.tf

```diff
@@ -1,3 +1,7 @@
 output "cumulus_message_adapter_layer_arn" {
```

Comment:
Thanks so much for this contribution @elkirby !

I know that we'd be breaking with the Terraform conventions, but maybe we call this output `cumulus_message_adapter_layer_version_arn` and the one below it `cumulus_message_adapter_layer_arn`. Avoids the repetitive names and is maybe clear about which output contains the version.

Alternately, do we even need an output for the ARN without the version? It seems like anything using this module needs the layer ARN with the version in it for compatibility with the Cumulus module.

Thoughts?

Reply:
No problem -- happy to contribute! I'm new to the whole project and am totally fine with whatever people decide makes the most sense. 

I personally like the naming that includes the word version, and will admit I didn't see anywhere that the version wasn't also required. 

----
Finally, and this is a bit beyond the scope of this PR, but, since we're talking naming, I thought I'd still bring it up for discussion... 😅 

Before writing this issue, I found #1361, which talks about not knowing whether to specify version or not for `cumulus_message_adapter_lambda_layer_arn` in the cumulus module. Your `cumulus_message_adapter_layer_version_arn` suggestion got me wondering if maybe applying this naming in both modules would solve that issue, in addition to keeping the names consistent and making it obvious that they're the same value. I'll admit that I too got tripped up originally by the issue outlined in #1361, and even when writing the linked issue for this PR, the description had to be edited because I originally got their names wrong.

Reply:
I like the `cumulus_message_adapter_layer_version_arn` naming, and it's nice that it would also address #1361. It would be a breaking change for deployment configuration, albeit a very minor one. Let's see what the team thinks.

@Jkovarik @laurenfrederick @nemreid - thoughts?

Reply:
@elkirby We discussed this as a team and we'd like to move forward with this approach:

- Change the output for the layer version ARN to `cumulus_message_adapter_lambda_layer_version_arn` in `tf-modules/cumulus_message_adapter/outputs.tf`. Remove the non-layer version output.
- Change the variable name for the layer from `cumulus_message_adapter_lambda_layer_arn` to `cumulus_message_adapter_lambda_layer_version_arn` in both `tf-modules/cumulus/variables.tf` and `tf-modules/ingest/variables.tf`

Would you be up for making those changes?

Reply:
@markdboyd, 
I would absolutely be up to it. 

Restating my understanding, just to be sure:
1. in cumulus-message-adapter/outputs.tf, there should be **one** output: a variable called `cumulus_message_adapter_lambda_layer_version_arn`, whose output should be `aws_lambda_layer_version.cumulus_message_adapter.arn` (that is, the ARN with the version appended)
2. Change ***all*** references of `cumulus_message_adapter_lambda_layer_arn` to `cumulus_message_adapter_lambda_layer_version_arn` in the following modules:
2a. `cumulus`
2b. `ingest`

Am I on the same page?

Does this cause downstream changes to other repositories that I should be aware of? I know cumulus-template-deploy [does not currently have a direct hook-up](https://github.com/nasa/cumulus-template-deploy/blob/master/cumulus-tf/main.tf#L43) between `cumulus-message-adapter` and `cumulus` child modules (which I have some ideas about how that could be done ...for another time 😅), but are there others I should be aware of that I could/should submit a contribution to?

Reply:
@elkirby Yep, your understanding is correct. Would you also mind updating `example/cumulus-tf/variables.tf` to replace `cumulus_message_adapter_lambda_layer_arn` with `cumulus_message_adapter_lambda_layer_version_arn`? This file is only used internally by the Cumulus core team, but it would be helpful to us 😄 

In terms of other repos, yeah the only other one would be cumulus-template-deploy, where we would want to update all instances of `cumulus_message_adapter_lambda_layer_arn` to `cumulus_message_adapter_lambda_layer_version_arn`. Thanks so much!

Reply:
✅ Done!
Everything is still split up into very small commits to make it clear what I did, and to make it easy to drop anything that y'all didn't want. I saw a couple other references to the variable, so once I validated that it made sense to do so, I added commits for each of those changes too. I will annotate them. Feel free to request other revisions, additions, or redactions.

Reply:
This is great @elkirby! Thanks for catching all those references that I didn't even mention. Going to merge this now. 💯 awesome contribution!

Reply:
Also, just to shed some light on our PR merge process from forks: we run our test suite on all PRs before merge as a blocking step, but we don't run any tests PRs from forks for security reasons, so I have to merge your work to an intermediary branch off of `master` and then merge that branch to `master`. And that PR is open now:

https://github.com/nasa/cumulus/pull/1977